intra_process_demo/CMakeLists.txt

```diff
@@ -132,6 +132,11 @@ if(BUILD_TESTING)
         TIMEOUT 30)
       set_tests_properties(test_demo_${exe_list_underscore}${target_suffix}
         PROPERTIES DEPENDS "test_demo_${exe_list_underscore}${target_suffix} test_demo_${exe_list_underscore}${target_suffix}")
+
+      set_property(
+        TEST test_demo_${exe_list_underscore}${target_suffix}
+        APPEND PROPERTY
+        ENVIRONMENT RCL_ASSERT_RMW_ID_MATCHES=${rmw_implementation})
```

Comment:
yeah it works fine. I chose it because of the append capability.. I figured it would interfere the least with any future code restructuring this way